lib/elixir/lib/inspect.ex

```diff
@@ -307,11 +307,18 @@ end
 
 defimpl Inspect, for: Regex do
   def inspect(regex, opts) do
-    {escaped, _} = Identifier.escape(regex.source, ?/, :infinity, &escape_map/1)
+    {escaped, _} =
+      regex.source
+      |> :elixir_interpolation.unescape_chars(&unescape_map/1)
```

Comment:
Unless I am missing something, isn't this the same as:

```suggestion
      |> String.replace("/", "\\/")
```

Reply:
It would be the other way around, replacing `"\\/"` with `"/"`, but only for single backslashes `"\\"`:

```elixir
iex(1)> :elixir_interpolation.unescape_chars("a\\/b", fn ?/ -> ?/; _ -> false end)
"a/b"
iex(2)> :elixir_interpolation.unescape_chars("a\\\\/b", fn ?/ -> ?/; _ -> false end)
"a\\\\/b"
iex(3)> :elixir_interpolation.unescape_chars("a\\\\\\/b", fn ?/ -> ?/; _ -> false end)
"a\\\\/b"
iex(4)> String.replace("a\\/b", "\\/", "/")
"a/b"
iex(5)> String.replace("a\\\\/b", "\\/", "/")
"a\\/b"
iex(6)> String.replace("a\\\\\\/b", "\\/", "/")
"a\\\\/b"
```

Reply:
Yes, I see what you mean. I would do a loop by hand then, using unescape_chars may be a bit too much. Thoughts?

Reply:
I’ll give it a try 🙂